lib/chef/resource/build_essential.rb

```diff
@@ -61,18 +61,17 @@ class BuildEssential < Chef::Resource
           package "devel/m4"
           package "devel/gettext"
         when macos?
-          unless xcode_cli_installed?
+          update_pkg_label = xcode_cli_package_label
```

Comment:
Since running `softwareupdate -l` is so slow, you could potentially save time in some cases by moving this method call inside the `unless` block on the next line.

Reply:
I think the current behavior is also acting as an updater, which is wrong. We should have an :upgrade action that handles that scenario.